pytket/pytket/qasm/qasm.py

```diff
@@ -1062,7 +1084,7 @@ def circuit_to_qasm_io(
             raise QASMUnsupportedError(
                 "Cannot print command of type: {}".format(op.get_name())
             )
-        if opstr not in include_gate_defs:
+        if opstr not in include_gate_defs and optype != OpType.Barrier:
```

Comment:
No need for the extra condition: barriers are already handled.

Reply:
This is required for the check not to fail for replacements done. I could add more conditions if you want?

Reply:
I guess it should be:
```python
        if opstr not in include_gate_defs and not (optype == OpType.Barrier and header == "hqslib1_dev"):
```
or equivalent?

Reply:
I would suggest to change this with a special flag to make sure that this check still works for classic barriers without data, too.